libraries/pulse/src/publisher.js

```diff
@@ -195,7 +195,19 @@ class PulsePublisher {
     this._setChannel(null);
     this.stopHandlingConnections = this.client.onConnected(this._handleConnection);
 
-    await this._assertExchanges();
+    // these tasks can stall waiting for a rabbitmq connection, so we log when
+    // they begin and end, and if they take too long
+
+    this.client.monitor.info('Asserting pulse exchanges exist');
+    const msg = 'Still attempting to assert pulse exchanges; is RabbitMQ configuration correct?';
+    const interval = setTimeout(() => this.client.monitor.warning(msg), 30000);
```

Comment:
What about `setInterval` for this?

Reply:
Whoops, that's what I meant (that's why I used `interval`)!